src/commands/SinceCommand.ts

```diff
@@ -244,7 +245,26 @@ async function execSinceCommandAux(destinationRoomId: string, event: any, mjolni
                             results.bad += 1;
                         }
                     }
-                    const text_ = `Attempted to ban ${joins.length} users from room ${targetRoomId}, ${results.good} kicked, ${results.bad} failures`;
+                    const text_ = `Attempted to ban ${joins.length} users from room ${targetRoomId}, ${results.good} banned, ${results.bad} failures`;
+                    return {
+                        html: text_,
+                        text: text_
+                    }
+                }
+                case Action.Mute: {
+                    const joins = mjolnir.roomJoins.getUsersInRoom(targetRoomId, minDate, maxEntries);
+
+                    let results = { good: 0, bad: 0};
+                    for (let join of joins) {
+                        try {
+                            await mjolnir.client.setUserPowerLevel(join.userId, targetRoomId, -1);
```

Comment:
This has the potential to cause a lot of state event churn when you also consider that setting the power level for a user edits just one state event `m.room.power_levels`. It is possible to change mute all of the users at the same time, but that might take a fair amount of extra code to get right and it might not be a feature of `matrix-bot-sdk`. Either way we should make a note of why we have not attempted to edit the event just once.

Reply:
Alright, rewrote this into a single rewrite.